src/control/Control.cpp

```diff
@@ -1623,7 +1623,7 @@ void Control::undoRedoChanged() {
 
 void Control::undoRedoPageChanged(PageRef page) {
     if (!this->changedPages.empty() &&
-        std::find(begin(this->changedPages), end(this->changedPages), page) == end(this->changedPages)) {
+        std::find(begin(this->changedPages), end(this->changedPages), page) != end(this->changedPages)) {
         return;
     }
 
```

Comment:
Wouldn't this be better if `Control::changedPages` was an std::unordered_set to begin with?
I may be wrong here, with `changedPages` being usually very small.

Reply:
I guess using `std::unordered_set` would be cleaner, but performance-wise I don't think it makes a big difference.

Reply:
We may even make things worse performance-wise (because of the hash computations and the allocation overheads). Let's leave it like that.